crates/swift-bridge-ir/src/codegen/codegen_tests/c_ffi_declaration_order_codegen_tests.rs

```diff
@@ -0,0 +1,55 @@
+use super::{CodegenTest, ExpectedCHeader, ExpectedRustTokens, ExpectedSwiftCode};
+use proc_macro2::TokenStream;
+use quote::quote;
+
+/// Verify that fields of a structure in a C header file are generated before the structure.
+mod structure_fields_generated_before_structure {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod ffi {
+                extern "Rust" {
+                    type ResultTestOpaqueRustType;
+                }
+
+                extern "Rust" {
+                    fn rust_func_return_result_tuple_transparent_enum(
+                        succeed: bool,
+                    ) -> Result<(i32, ResultTestOpaqueRustType, String), i32>;
```

Comment:
Maybe `Result<(RustOpaqueType1, RustOpaqueType1), (RustOpaqueType2, RustOpaqueType2)>` ?
(I can't remember if we allow tuples in the `E` of `Result<T, E>`)

Reply:
I don't implement it, but it might work.

Here's a PR which adds support `Result<tuple, _>`
https://github.com/chinedufn/swift-bridge/pull/213